script.js

```diff
@@ -0,0 +1,521 @@
+
+    if (char === '&lt;') { char = '<'; }
+    if (char === '&gt;') { char = '>'; }
+
+    textarea.value = [beforeCursor, char, afterCursor].join('');
+
+    textarea.selectionStart += 1;
+    textarea.selectionEnd += 1;
+  } else if (e.target.classList.contains('active')
+    && !functionalKeys.includes(e.target.classList[1])) {
+    cursorStart = textarea.selectionStart;
+    cursorEnd = textarea.selectionEnd;
+
+    const beforeCursor = textarea.value.slice(0, cursorStart);
+    const afterCursor = textarea.value.slice(cursorStart, textarea.value.length);
+    let char = e.target.innerHTML;
+
+    if (char === '&lt;') { char = '<'; }
+    if (char === '&gt;') { char = '>'; }
+
+    textarea.value = [beforeCursor, char, afterCursor].join('');
+
+    textarea.selectionStart = cursorStart + 1;
+    textarea.selectionEnd = cursorEnd + 1;
+  }
+
+  if (e.target.classList.contains('key')
+    && !(e.target.classList.contains('CapsLock')
+        || e.target.parentElement.parentElement.classList.contains('CapsLock'))) {
+    e.target.classList.add('pressed');
+  } else if (e.target.classList.contains('active')
+    && !(e.target.classList.contains('CapsLock')
+        || e.target.parentElement.parentElement.classList.contains('CapsLock'))) {
+    e.target.parentElement.parentElement.classList.add('pressed');
+  }
+});
+
+document.addEventListener('mouseup', (e) => {
+  textarea.focus();
+
+  if (e.target.classList.contains('ShiftLeft')
+    || e.target.parentElement.parentElement.classList.contains('ShiftLeft')
+    || e.target.classList.contains('ShiftRight')
+    || e.target.parentElement.parentElement.classList.contains('ShiftRight')) {
+    doUnshift();
+  }
+  if (e.target.classList.contains('key')
+    && !(e.target.classList.contains('CapsLock')
+        || e.target.parentElement.parentElement.classList.contains('CapsLock'))) {
+    e.target.classList.remove('pressed');
+  } else if (e.target.classList.contains('active')
+    && !(e.target.classList.contains('CapsLock')
+        || e.target.parentElement.parentElement.classList.contains('CapsLock'))) {
+    e.target.parentElement.parentElement.classList.remove('pressed');
+  }
+  if (!e.target.classList.contains('Tab')
+    && !e.target.parentElement.classList.contains('Tab')
+    && !e.target.classList.contains('Delete')
+    && !e.target.parentElement.classList.contains('Delete')) {
+    textarea.selectionStart = cursorStart + 1;
+    textarea.selectionEnd = cursorEnd + 1;
+  }
+});
+
+// key presses
+document.addEventListener('keydown', (e) => {
+  e.preventDefault();
+  textarea.focus();
+  if (functionalKeys.includes(e.code)) {
+    document.querySelector(['.', e.code].join('')).classList.add('pressed');
+  } else {
+    document.querySelector(['.', e.code].join('')).parentElement.classList.add('pressed');
+  }
+
+  if ((e.code === 'ShiftRight' && (e.ctrlKey))
+    || (e.code === 'ShiftLeft' && (e.ctrlKey))) {
```

Comment:
seems like condition (e.ctrlKey) can be checked once, and not twice, like here: 
 if ((e.code === 'ShiftLeft' || e.code === 'ShiftRight') && !e.ctrlKey)